File: app/src/main/java/com/example/myfakequotesapp/View/Adapters/FilmResultsAdapter.java
package com.example.myfakequotesapp.View.Adapters;

import android.net.Uri;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.myfakequotesapp.Model.Quote;
import com.example.myfakequotesapp.R;
import com.example.myfakequotesapp.pojo.Result;
import com.squareup.picasso.Picasso;

import java.util.ArrayList;
import java.util.List;

import static android.content.ContentValues.TAG;

public class FilmResultsAdapter extends RecyclerView.Adapter<FilmResultsAdapter.FilmResultsViewHolder> {
    public OnPosterClickListener getOnPosterClickListener() {
        return onPosterClickListener;
    }

    private List<Result> films;
    private OnPosterClickListener onPosterClickListener;
    private FilmResultsAdapter.OnPosterClickListener listener;

    public FilmResultsAdapter() {
        this.films = new ArrayList<>();
    }

    public interface OnPosterClickListener {
        void onPosterClick(int position);
    }

    public void setOnPosterClickListener(OnPosterClickListener onPosterClickListener) {
        this.onPosterClickListener = onPosterClickListener;
    }

    public List<Result> getFilms() {
        return films;
    }

    public void setFilms(List<Result> films) {
        this.films = films;
        Log.d(TAG,"IN CONSTRUCTOR: "+films.get(0).getImage());
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public FilmResultsAdapter.FilmResultsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.film_results_item, parent, false);
        return new FilmResultsViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull FilmResultsViewHolder holder, int position) {

        Result film = films.get(position);
        Picasso.get().load(film.getImage()).into(holder.imageViewSmallPoster);
        Log.d(TAG, film.getImage());
        Log.d(TAG,"OnBindViewHolder: " + film.getImage());
    }

    @Override
    public int getItemCount() {
        return films.size();
    }

    class FilmResultsViewHolder extends RecyclerView.ViewHolder {
        private ImageView imageViewSmallPoster;

        public FilmResultsViewHolder(@NonNull View itemView) {
            super(itemView);

            imageViewSmallPoster = itemView.findViewById(R.id.imageViewSmallPoster);
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (onPosterClickListener != null) {
                        onPosterClickListener.onPosterClick(getAdapterPosition());
                    }
                }
            });
        }
    }


}

File: app/src/main/java/com/example/myfakequotesapp/View/Adapters/FilmResultsAdapter.kt
package com.example.myfakequotesapp.View.Adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.example.myfakequotesapp.R
import com.example.myfakequotesapp.View.Adapters.FilmResultsAdapter.FilmResultsViewHolder
import com.example.myfakequotesapp.pojo.Result
import com.squareup.picasso.Picasso

class FilmResultsAdapter : RecyclerView.Adapter<FilmResultsViewHolder>() {
    internal var films: List<Result?>?
    var onPosterClickListener: OnPosterClickListener? = null
    private val listener: OnPosterClickListener? = null

    interface OnPosterClickListener {
        fun onPosterClick(position: Int)
    }

    @JvmName("setOnPosterClickListener1")
    fun setOnPosterClickListener(onPosterClickListener: OnPosterClickListener?) {
        this.onPosterClickListener = onPosterClickListener
    }

    fun getFilms(): List<Result?>? {
        return films
    }

    fun setFilms(films: List<Result?>?) {
        this.films = films
        //Log.d(ContentValues.TAG, "IN CONSTRUCTOR: " + films!![0].getImage())
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FilmResultsViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.film_results_item, parent, false)
        return FilmResultsViewHolder(view)
    }

    override fun onBindViewHolder(holder: FilmResultsViewHolder, position: Int) {
        val film = films!![position]
        //Picasso.get().load(film.getImage()).into(holder.imageViewSmallPoster)
        Picasso.get().load(film?.image).into(holder.imageViewSmallPoster)
        //Log.d(ContentValues.TAG, film.getImage())
        //Log.d(ContentValues.TAG, "OnBindViewHolder: " + film.getImage())
    }

    override fun getItemCount(): Int {
        return films!!.size
    }

    inner class FilmResultsViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val imageViewSmallPoster: ImageView

        init {
            imageViewSmallPoster = itemView.findViewById(R.id.imageViewSmallPoster)
            itemView.setOnClickListener {
                if (onPosterClickListener != null) {
                    onPosterClickListener!!.onPosterClick(adapterPosition)
                }
            }
        }
    }

    init {
        films = ArrayList()
    }
}